flake.nix

```diff
@@ -32,13 +32,23 @@
           legacyPackages = devshell;
           devShell = devshell.fromTOML ./devshell.toml;
         };
+      fromTOML = path:  eachSystem (system:
+        let pkgs = import nixpkgs {
+          inherit system;
+          overlays = [ (import ./overlay.nix) ];
+        };
```

Comment:
Each instance of nixpkgs takes a minimum 30MB of RAM. If we can, it's better to use `nixpkgs.legacyPackages.${system}` since its evaluation can be shared between the various flakes (assuming they all follow the same version).

As a user, I would also like to potentially pass my own pkgs instance with my own overrides.

Reply:
Good point but I don't know how to do that
How to apply overlay to pkgs without import?

I tried this instead
```nix
{
  devShell = devshell.legacyPackages.${system}.fromTOML path;
}
```
but it failed in my project with: attribute 'nonEmptyStr' missing